src/Autofac/RegistrationExtensions.cs

```diff
@@ -69,6 +69,11 @@ public static IRegistrationBuilder<T, SimpleActivatorData, SingleRegistrationSty
 
         rb.SingleInstance();
 
+        // https://github.com/autofac/Autofac/issues/1102
+        // Single instance registrations with any custom activation phases (i.e. activation handlers) need to be auto-activated,
+        // so that other behavior (such as OnRelease) that expects 'normal' object lifetime behavior works as expected.
+        rb.RegistrationData.AddService(new AutoActivateService());
```

Comment:
This simplification wasn't possible prior to the other fix, as it would have broken default service logic.